sdk/storage/Azure.Storage.Files/src/Models/NtfsAttributes.cs

```diff
@@ -0,0 +1,137 @@
+        /// File systems try to keep all the data in memory for quicker access rather than
+        /// flushing the data back to mass storage. A temporary file should be deleted by
+        /// the application as soon as it is no longer needed.
+        /// </summary>
+        Temporary = FileAttributes.Temporary,
+
+        /// <summary>
+        /// The file or directory is offline. The data of the file is not immediately available.
+        /// </summary>
+        Offline = FileAttributes.Offline,
+
+        /// <summary>
+        /// The file or directory will not be indexed by the operating system's content indexing service.
+        /// </summary>
+        NotContentIndexed = FileAttributes.NotContentIndexed,
+
+        /// <summary>
+        /// The file or directory is excluded from the data integrity scan. When this value
+        /// is applied to a directory, by default, all new files and subdirectories within
+        /// that directory are excluded from data integrity.
+        /// </summary>
+        NoScrubData = FileAttributes.NoScrubData
+    }
+
+    /// <summary>
+    /// CloudFileNtfsAttributesHelper helper.
+    /// </summary>
+    internal class CloudFileNtfsAttributesHelper
+    {
+        private static readonly Dictionary<NtfsAttributes, string> dictionary = new Dictionary<NtfsAttributes, string>()
+        {
+            { NtfsAttributes.ReadOnly, "ReadOnly" },
+            { NtfsAttributes.Hidden, "Hidden" },
+            { NtfsAttributes.System, "System" },
+            { NtfsAttributes.Normal, "Normal" },
+            { NtfsAttributes.Directory, "Directory" },
+            { NtfsAttributes.Archive, "Archive" },
+            { NtfsAttributes.Temporary, "Temporary" },
+            { NtfsAttributes.Offline, "Offline" },
+            { NtfsAttributes.NotContentIndexed, "NotContentIndexed" },
+            { NtfsAttributes.NoScrubData, "NoScrubData" },
+        };
+
+        /// <summary>
+        /// Converts a CloudFileNtfsAttributes to a string
+        /// </summary>
+        /// <param name="attributes"><see cref="NtfsAttributes"/></param>
+        /// <returns>string</returns>
+        internal static string ToString(NtfsAttributes? attributes)
+            => attributes == null
+                ? null
+                : String.Join("|", dictionary.Select(
+                    r =>
+                    {
+                        return attributes.Value.HasFlag(r.Key) ? r.Value : null;
+                    })
+                .Where(r => r != null));
+        /// <summary>
+        /// Parses an attributes string to a <see cref="NtfsAttributes"/>
+        /// </summary>
+        /// <param name="attributesString">string</param>
+        /// <returns><see cref="NtfsAttributes"/></returns>
+        internal static NtfsAttributes? ToAttributes(string attributesString)
+        {
+            if (attributesString == null)
+            {
+                return null;
```

Comment:
return `None` instead?

Reply:
Nope.  None is a valid NTFS attribute, causing us to override the File/Directory's existing NTFS attributes when the user didn't intend us to.